packages/palette/src/elements/Modal/Modal.tsx

```diff
@@ -51,11 +55,34 @@ export const Modal: SFC<ModalProps> = ({
   })
   const [visibleContent, setVisibleContent] = useState(null)
 
-  const previousChildren = usePrevious(childrenState)
   const contentAnimation = useSpring(springContentAnimation)
   const modalAnimation = useSpring(springModalAnimation)
 
+  const handleEscapeKey = event => {
+    if (event && event.key === "Escape") {
+      onClose()
+    }
+  }
+
+  const updateEscapeKeyListener = () => {
+    if (show) {
+      document.addEventListener(KEYBOARD_EVENT, handleEscapeKey, true)
```

Comment:
Don't think this needs to be moved into a `KEYBOARD_EVENT` var. Rule of three's  / doesn't add much descriptive value.